server/src/main/java/org/elasticsearch/action/ingest/SimulatePipelineTransportAction.java

```diff
@@ -8,22 +8,48 @@
 
 package org.elasticsearch.action.ingest;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionListenerResponseHandler;
 import org.elasticsearch.action.support.ActionFilters;
 import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.node.DiscoveryNodes;
+import org.elasticsearch.common.Randomness;
 import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Setting;
 import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.core.TimeValue;
 import org.elasticsearch.ingest.IngestService;
 import org.elasticsearch.tasks.Task;
 import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportRequestOptions;
 import org.elasticsearch.transport.TransportService;
 
+import java.util.Collection;
 import java.util.Map;
+import java.util.Random;
 
 public class SimulatePipelineTransportAction extends HandledTransportAction<SimulatePipelineRequest, SimulatePipelineResponse> {
-
+    private static final Logger logger = LogManager.getLogger(SimulatePipelineTransportAction.class);
+    /**
+     * This is the amount of time given as the timeout for transport requests to the ingest node.
+     */
+    public static final Setting<TimeValue> INGEST_NODE_TRANSPORT_ACTION_TIMEOUT = Setting.timeSetting(
+        "ingest_node.transport_action_timeout",
+        TimeValue.timeValueSeconds(20),
+        TimeValue.timeValueMillis(1),
+        Setting.Property.NodeScope,
+        Setting.Property.Dynamic
+    );
     private final IngestService ingestService;
     private final SimulateExecutionService executionService;
+    private final TransportService transportService;
+    private TimeValue ingestNodeTransportActionTimeout;
```

Comment:
I think we usually make these `volatile`

Reply:
Oh good point -- I forgot that I made it so that it's updatable.